daemon/cmd/ciliumendpoints.go

```diff
@@ -0,0 +1,112 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/cilium/cilium/pkg/endpoint"
+	cilium_v2a1 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	ciliumv2 "github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/typed/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/types"
+	"github.com/cilium/cilium/pkg/node"
+)
+
+type endpointCache interface {
+	LookupPodName(name string) *endpoint.Endpoint
+}
+
+// Used by cleanStaleCEPs to retrieve ceps/pods to do cleanup of stale ceps.
+// Values retrieved only used as keys for accessing other caches and performing
+// deletes.
+func (d *Daemon) getStore(name string) cache.Store {
+	return d.k8sWatcher.GetStore(name)
+}
+
+// cleanStaleCEPs runs on Daemon init, attempting to delete any CiliumEndpoints that are referencing local
+// Pods that are not being managed (i.e. do not have Endpoints).
+// This must only be run after K8s Pod and CES/CEP caches are synced and local endpoint restoration is complete.
+func (d *Daemon) cleanStaleCEPs(ctx context.Context, eps endpointCache, ciliumClient ciliumv2.CiliumV2Interface, cepCRDEnabled bool) error {
+	keyFn := func(ns string, name string) string { return fmt.Sprintf("%s/%s", ns, name) }
```

Comment:
```suggestion
	keyFn := func(ns string, name string) string { return ns + "/" + name }
```

No need for `fmt.Sprintf`